kafka-ballerina/producer.bal

```diff
@@ -75,52 +76,13 @@ public client class Producer {
     # kafka:ProducerError? result = producer->send("Hello World, Ballerina", "kafka-topic");
     # ```
     #
-    # + value - Record contents
-    # + topic - Topic to which the record will be appended
-    # + key - Key, which will be included in the record
-    # + partition - Partition to which the record should be sent
-    # + timestamp - Timestamp of the record in milliseconds since epoch
+    # + producerRecord - Record to be produced
     # + return -  A `kafka:ProducerError` if send action fails to send data or else '()'
-    isolated remote function send(anydata value, string topic, anydata? key = (), int? partition = (),
-        int? timestamp = ()) returns ProducerError? {
-        // Handle string values
-        if (self.valueSerializerType == SER_STRING) {
-            if (value is string) {
-                return sendStringValues(self, value, topic, key, partition, timestamp, self.keySerializerType);
-            }
-            panic getValueTypeMismatchError(STRING);
-        }
-        // Handle int values
-        if (self.valueSerializerType == SER_INT) {
-            if (value is int) {
-                return sendIntValues(self, value, topic, key, partition, timestamp, self.keySerializerType);
-            }
-            panic getValueTypeMismatchError(INT);
-        }
-        // Handle float values
-        if (self.valueSerializerType == SER_FLOAT) {
-            if (value is float) {
-                return sendFloatValues(self, value, topic, key, partition, timestamp, self.keySerializerType);
-            }
-            panic getValueTypeMismatchError(FLOAT);
-        }
-        // Handle byte[] values
+    isolated remote function sendProducerRecord(ProducerRecord producerRecord) returns ProducerError? {
```

Comment:
Do we need to change the function name here?

Reply:
All other messaging modules' `send()` functions were changed to `sendMessage(Message message)`. The idea was to have that as the default case and add methods for sending other types as `sendStringMessage()` and so on. So in Kafka we changed `send` to `sendProducerRecord(ProducerRecord record)` and `onMessage` to `onConsumerRecord`.